### ISSUE TITLE
The binary version of its metadata is 1.8.0, expected version is 1.6.0.

### ISSUE THREAD

**Issue by Nasseratic** [hirbod/react-native-volume-manager#18]
Hi 👋🏼 

I am getting this error trying to build android with expo 49 ( react native 0.72 )

Maybe upgrade the Kotlin version again like this issue https://github.com/hirbod/react-native-volume-manager/issues/11
```
> Task :react-native-volume-manager:compileDebugKotlin FAILED

[stderr] 
e: Incompatible classes were found in dependencies. Remove them from the classpath or use '-Xskip-metadata-version-check' to suppress errors

[stderr] 
e: /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
e: /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
e: /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
e: /home/expo/.gradle/caches/transforms-3/f8b9e1ac9c843d8dc990942237200df0/transformed/jetified-kotlin-stdlib-common-1.8.10.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/Utils.kt: (7, 20): Unresolved reference: ln

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/Utils.kt: (22, 10): Not enough information to infer type variable T

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/Utils.kt: (22, 76): Unresolved reference: java

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/Utils.kt: (38, 30): Unresolved reference: ln

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/Utils.kt: (39, 13): Unresolved reference: ln

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (22, 12): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (23, 12): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (24, 13): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (45, 5): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (46, 15): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (48, 15): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (55, 12): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (56, 12): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class

[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-volume-manager/android/src/main/java/com/reactnativevolumemanager/VolumeManagerSilentListenerModule.kt: (62, 29): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/0a487ac5fce194b44f47bee930b70ae9/transformed/jetified-kotlin-stdlib-1.8.10.jar!/kotlin/Unit.class
```

**Comment by Nasseratic** [comment]
I got it working by applying this patch: 
```
diff --git a/android/build.gradle b/android/build.gradle
index a5cbedf0a81e90b9dba4c617fe109144a9c7b7e8..ecc8e04aad96469d8012b215edfce2d769e75b21 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,6 +1,6 @@
 buildscript {
   // Buildscript is evaluated before everything else so we can't use getExtOrDefault
-  def kotlin_version = "1.6.20"
+  def kotlin_version = "1.8.0"
 
   repositories {
     google()
diff --git a/android/gradle.properties b/android/gradle.properties
index 10376a6dccf28bee05f74c1a5e317e893007d2f6..17b35ad480b7db41804718af9e3a72f82449145b 100644
--- a/android/gradle.properties
+++ b/android/gradle.properties
@@ -1,4 +1,4 @@
-VolumeManager_kotlinVersion=1.6.20
+VolumeManager_kotlinVersion=1.8.0
 VolumeManager_compileSdkVersion=33
 VolumeManager_targetSdkVersion=33
 
 ```
 
 Will open a pr soon :)

**Comment by hirbod** [comment]
Fixed with #19, will release a new version. Thanks for your contribution
